app/ai/voice/agents/breeze_buddy/template/loader.py

```
@@ -87,13 +96,90 @@ def render_task_messages(
+        # 1. Redis cache check (requires lead_id)
+        if lead_id:
+            cache_key = f"datasource:{lead_id}:{ref.name}"
+            try:
+                redis = await get_redis_service()
+                cached = await redis.get(cache_key)
+                if cached:
+                    logger.info(
+                        "Data source cache hit: lead=%s name=%s", lead_id, ref.name
+                    )
+                    return cached
+            except Exception as exc:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)